src/Features/Core/Portable/MoveToNamespace/MoveToNamespaceResult.cs

```diff
@@ -8,12 +10,21 @@ internal class MoveToNamespaceResult
 
         public bool Succeeded { get; }
         public Solution UpdatedSolution { get; }
+        public Solution OriginalSolution { get; }
         public DocumentId UpdatedDocumentId { get; }
+        public IReadOnlyDictionary<string, ISymbol> NewNameOriginalSymbolMapping { get; }
```

Comment:
`ImmutableDictionary<string, ISymbol>` instead

Reply:
What's the benefit over `IReadOnlyDictionary`?

Reply:
Dictionary implements IReadOnlyDictionary, but that does not ensure it's actually read-only, so it can be mutated by external means (i.e. caller passes it somewhere else as `Dictionary<T>`)

On the other hand, faster enumeration, since concrete types provide struct enumerators compared to the boxed enumerator coming in from the `IEnumerable<T>`

Reply:
Awesome, thanks for the explanation. I changed to `ImmutableDictionary`